api/librdb-ext-api.h

```diff
@@ -25,6 +25,9 @@ typedef enum {
     RDBX_ERR_READER_FILE_GENERAL_ERROR = _RDB_ERR_EXTENSION_FIRST,
     RDBX_ERR_RESP_FAILED_ALLOC,
 
+    /* reader errors */
+    RDBX_ERR_FAILED_FD_OPEN,
```

Comment:
considering that the API and maybe even ABI should be stable across versions, i'm not sure if we want the error to be that specific, and if being that specifc somehow serves the caller at all.
why not use RDBX_ERR_FAILED_OPEN_FILE?
i.e. the failure happened while preparing the file (descriptor) for reading.